x/logic/predicate/file.go

```diff
@@ -2,12 +2,13 @@
 
 import (
 	"context"
-	"fmt"
 	"os"
 	"reflect"
 	"sort"
 
 	"github.com/ichiban/prolog/engine"
+
+	"github.com/okp4/okp4d/x/logic/prolog"
 )
 
 // SourceFile is a predicate that unify the given term with the currently loaded source file.
```

Comment:
Yes, good point. This code predates the changes introduced in this PR, but indeed, it's important to address it correctly.

Reply:
@coderabbitai, could you open an issue regarding this matter?

Reply:
@ccamel @coderabbitai, it seems inevitable to avoid using reflection to me.. It would require to access this VM property without, and in my sense we shouldn't be able to do that.